### DISCUSSION TITLE
Maybe some trouble with the `run_sketch` helper?

### DISCUSSION THREAD

**Discussion by villares** [py5coding/py5generator#371] [Q&A] [answered]
I've been using Thonny IDE for years now with @tabreturn's plug-in for my own sketches and for teaching.

Today I was wondering which were the `run_sketch` (command line) options and if they were properly documented somewhere, so I decided to try it and was surprised. Something is not right in my machine, I tried reinstalling *py5* and *py5jupyter* but it didn't help:
<pre><font color="#00CD00"><b>[villares@L340</b></font><font color="#AFA5A5"><b> sketch_2023_10_16</b></font><font color="#00CD00"><b>]$</b></font> conda activate py5coding
(py5coding) <font color="#00CD00"><b>[villares@L340</b></font><font color="#AFA5A5"><b> sketch_2023_10_16</b></font><font color="#00CD00"><b>]$</b></font> run_sketch sketch_2023_10_16.py
Traceback (most recent call last):
  File &quot;/home/villares/.local/bin/run_sketch&quot;, line 5, in &lt;module&gt;
    from py5_tools.tools.run_sketch import main
ModuleNotFoundError: No module named &apos;py5_tools&apos;
(py5coding) <font color="#00CD00"><b>[villares@L340</b></font><font color="#AFA5A5"><b> sketch_2023_10_16</b></font><font color="#00CD00"><b>]$</b></font> python
Python 3.10.12 | packaged by conda-forge | (main, Jun 23 2023, 22:40:32) [GCC 12.3.0] on linux
Type &quot;help&quot;, &quot;copyright&quot;, &quot;credits&quot; or &quot;license&quot; for more information.
&gt;&gt;&gt; import py5_tools
&gt;&gt;&gt; import py5
&gt;&gt;&gt; 
</pre>

I'll investigate more and post here. It seems like the `run_sketch` being called is outside the *conda env* and being executed on another python interpreter, *ooff*; I hope students don't get messy multiple pythons on their computers (but I'm afraid they will).

Update question: Who added `"/home/villares/.local/bin/run_sketch` to my system? It has a `#!` pointing to another python. I tried removing it (actually moving it somewhere else for a while) and `pip install py5 --force-reinstall py5` but it didn't get recreated.

**Comment by villares** [comment, marked as answer]
Well, "fixing" the `#!` of my `run_sketch.py` I could make it work. Now, maybe a documentation issue. Could we have a brief page for the `run_sketch` tool on py5coding.org? I could only find a reference to the `py5.run_sketch()` method: https://py5coding.org/search.html?q=run_sketch 

There could be a link to it at https://py5coding.org/content/py5_modes.html#imported-mode

```
usage: run_sketch [-h] [-c CLASSPATH] [--py5_options [PY5_OPTIONS ...]]
                  [--sketch_args [SKETCH_ARGS ...]]
                  sketch_path

Execute py5 sketch

positional arguments:
  sketch_path           path to py5 sketch

options:
  -h, --help            show this help message and exit
  -c CLASSPATH, --classpath CLASSPATH
                        extra directories to add to classpath
  --py5_options [PY5_OPTIONS ...]
                        list of parameters to pass to Processing (do not
                        prefix anything with a "-")
  --sketch_args [SKETCH_ARGS ...]
                        list of parameters to pass to py5 (do not prefix
                        anything with a "-")
```

**Reply by hx2A** [reply]
Hmmm, this was a strange issue. Somehow the py5 library was installed but `run_sketch` was installed with a corrupt file? I don't understand what would have caused that. In any case, I'm glad you were able to fix it. I would have tried recreating the whole py5 environment if what you did didn't work.

I agree, `run_sketch` should be documented somewhere. Without documentation, people won't know about it and won't be inspired to create innovative things like the thonny plugin!

**Reply by villares** [reply]
Another nice thing would be to have some more info/usage tips about the `py5_options` and the `sketch_args` ....

**Reply by hx2A** [reply]
> Another nice thing would be to have some more info/usage tips about the py5_options and the sketch_args ....

Yes, agreed. Currently every time I want to know what they are I dive into the Processing source code.
